cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -752,12 +752,12 @@ newtype CostModel = CostModel (Map Text Integer)
 validateCostModel :: PlutusScriptVersion lang
                   -> CostModel
                   -> Either InvalidCostModel ()
-validateCostModel PlutusScriptV1 (CostModel m)
-  | Alonzo.isCostModelParamsWellFormed m = Right ()
-  | otherwise                        = Left (InvalidCostModel (CostModel m))
-validateCostModel PlutusScriptV2 (CostModel m)
-  | Alonzo.isCostModelParamsWellFormed m = Right ()
-  | otherwise                        = Left (InvalidCostModel (CostModel m))
+validateCostModel PlutusScriptV1 (CostModel m) =
+    bimap (\_err -> InvalidCostModel (CostModel m)) id
```

Comment:
do you maybe want to propagate the error?

Reply:
This is a fair question. My goal was this PR was to unblock the dependency bumps.

cc @Jimbo4350 re propagating this error now?

Reply:
Yeah we may as well do it now. I can take over.